astropy/units/decorators.py

```diff
@@ -250,12 +308,24 @@ def wrapper(*func_args, **func_kwargs):
             with add_enabled_equivalencies(self.equivalencies):
                 return_ = wrapped_function(*func_args, **func_kwargs)
 
-            valid_empty = (inspect.Signature.empty, None)
-            if wrapped_signature.return_annotation not in valid_empty:
-                return return_.to(wrapped_signature.return_annotation)
-            else:
+            # Return
+            # if there is no annotation, return as-is
+            valid_empty = (inspect.Signature.empty, None, NoneType)
+            if wrapped_signature.return_annotation in valid_empty:
                 return return_
 
+            # annotated output
+            ret_targets = _parse_annotation(
+                wrapped_signature.return_annotation)
+            # TODO for future PR: drop support for Quantity when mean Unit
```

Comment:
what does "when mean Unit" mean?

Reply:
``Unit(Quantity(...))`` produces a unit. Currently ``quantity_input`` accepts a ``Quantity`` annotation since it will be turned into a unit. I don't think this was intentional and should probably be deprecated. If someone wants to annotate for units of, e.g. ``2 * u.km`` then they should explicitly use ``Unit(2 * u.km)``